src/Mvc/Mvc.TagHelpers/src/CacheTagHelperMemoryCacheFactory.cs

```diff
@@ -22,6 +29,9 @@ internal CacheTagHelperMemoryCacheFactory(IMemoryCache cache)
             Cache = cache;
         }
 
+        /// <summary>
+        /// A shared <see cref="IMemoryCache"/> used by all <see cref="CacheTagHelper"/>s.
```

Comment:
The entire purpose of this class is to be registered by DI for use  by `CacheTagHelper`. Hence the name of the class `CacheTagHelperMemoryCacheFactory`.  I'd rather users be informed of those implications rather than re-purpose the class risking unintended consequences.

To avoid blocking on this I'll change the summary to something more generic `Gets the <see cref="IMemoryCache"/>.`